dev/NavigationView/NavigationView.cpp

```diff
@@ -1494,7 +1500,25 @@ void NavigationView::UpdatePaneLayout()
                         {
                             const auto footersActualHeight = footerItemsRepeater.ActualHeight();
                             const auto menuItemsActualHeight = menuItems.ActualHeight();
-                            if (totalAvailableHeight > menuItemsActualHeight + footersActualHeight)
+
+                            if (m_footerItemsSource.Count() == 0 && !IsSettingsVisible())
+                            {
+                                footerItemsScrollViewer.MaxHeight(0);
```

Comment:
I think it would be better to add a new state to the NavigationView's VSM and call GoToState, instead of setting these layout properties from code.

Reply:
Added a VisualState for the separator, however left the max height setting as is since I am not sure how overriding the VisualState's setter would work and I want to prevent ending in a bad state.

Reply:
@chingucoding I don't think we need maxHeight setter for the bottom footer. If there are no items - it naturally will have height 0 :)

Reply:
Good point, removed that setter.